api/docs/reference/search_algorithm.md

```diff
@@ -0,0 +1,226 @@
+
+1. General "query" searching, which applies a simple text query search against
+   several, non-configurable fields.
+2. Individual field querying, which enables searching specific fields with
+   independent query terms.
+
+Both of these search types also allow further keyword filtering along the
+following fields:
+
+- Extension
+- Category
+- Length
+- Aspect ratio
+- Size
+- Source
+- License
+
+Source is the only field for which you can currently also specify exclusions.
+
+By default, items marked "mature" are excluded, but these can also be enabled.
+
+See the following API documentation links for descriptions and options for each
+field:
+
+- [Audio search](https://api.openverse.engineering/v1/#operation/audio_search)
+- [Image search](https://api.openverse.engineering/v1/#operation/image_search)
+
+Each of these fields are searched relatively strictly, primarily because the
+search domain in each is very small and "keyword" like. That is, there is a
+limited and specific set of terms that appear for the relevant document fields
+for each of these query parameters. All of them are validated to only allow
+specific options (documented in the API documentation links above), which
+enforces the "keyword" like nature of their usage.
+
+### General "query" searching
+
+This is the type of searching most commonly used to query Openverse. It is
+activated when the `q` query parameter is present. It searches the following
+aspects of a document:
+
+- Description
+- Title
+- Tags
+
+> **Note**: The API does not currently surface the "description" field, though
+> [discussion is underway](https://github.com/WordPress/openverse-catalog/issues/364)
+> to potentially change this fact.
+
+Of these, title is weighted 10000 times more heavily than the description and
+tags. This makes searches that match a title very closely rise to the "top" of
+the results, even if the same text is present word-for-word in a description. It
+also breaks ties between documents, if for example two documents are returned,
+one because the title matches and one because a tag matches, the title-matched
+document will be ranked higher and therefore appear first.
+
+Additionally, if the query is wrapped in double quotes (`"`), we search each of
+these _exactly_ which will bypass stemming and match exact word order. Title
+weighting is still applied in this case. This means that if the exact query text
+is found in the title of one document and the description of another, the
+title-matched document will appear ahead of the description-matched document.
+
+### Individual field querying
+
+This type of search is not commonly used. The Openverse frontend only surfaces
+one of the subset of fields that are queryable in this approach. It is only
+available if the `q` query parameter is excluded. In other words, even if the
+request includes these query parameters, _the API ignores these options if the
+`q` query parameter is present_. In that case, it will execute the
```

Comment:
```suggestion
available if the `q` query parameter is excluded. In other words, if the
request includes the `q` query parameters, _the API will ignore these options
even if they are present in the request_. In that case, it will execute the
```

Reply:
Much clearer, thank you.